mzLib/MzLibUtil/ClassExtensions.cs

```diff
@@ -135,5 +135,70 @@ public static string GetPeriodTolerantFilenameWithoutExtension(this string fileP
             return PeriodTolerantFilenameWithoutExtension.GetPeriodTolerantFilenameWithoutExtension(filePath);
         }
 
+        /// <summary>
+        /// Determines whether the collection is null or contains no elements.
+        /// </summary>
+        /// <typeparam name="T">The IEnumerable type.</typeparam>
+        /// <param name="enumerable">The enumerable, which may be null or empty.</param>
+        /// <returns>
+        ///     <c>true</c> if the IEnumerable is null or empty; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsNullOrEmpty<T>(this IEnumerable<T> enumerable)
+        {
+            if (enumerable == null)
+            {
+                return true;
+            }
+            /* If this is a list, use the Count property.
+             * The Count property is O(1) while IEnumerable.Count() is O(N). */
+            if (enumerable is ICollection<T> collection)
+            {
+                return collection.Count < 1;
+            }
+            return !enumerable.Any();
+        }
+
+        /// <summary>
+        /// Determines whether the collection is null or contains no elements.
+        /// </summary>
+        /// <typeparam name="T">The IEnumerable type.</typeparam>
+        /// <param name="collection">The collection, which may be null or empty.</param>
+        /// <returns>
+        ///     <c>true</c> if the IEnumerable is null or empty; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsNullOrEmpty<T>(this ICollection<T> collection)
+        {
+            if (collection == null)
+            {
+                return true;
+            }
+            return collection.Count < 1;
+        }
+
+        /// <summary>
+        /// Determines whether the dictionary is null or contains no elements.
+        /// </summary>
+        /// <param name="dictionary">The dictionary, which may be null or empty.</param>
+        /// <returns>
+        ///     <c>true</c> if the dictionary is null or empty; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsNullOrEmpty<TKey, TValue>(this IDictionary<TKey, TValue> dictionary)
+        {
+            return dictionary == null || dictionary.Count < 1;
+        }
+
+        /// <summary>
+        /// Determines whether the dictionary is null or contains no elements.
+        /// </summary>
+        /// <param name="dictionary">The dictionary, which may be null or empty.</param>
+        /// <returns>
+        ///     <c>true</c> if the dictionary is null or empty; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsNotNullOrEmpty<TKey, TValue>(this IDictionary<TKey, TValue> dictionary)
+        {
```

Comment:
This is already defined in Easy.Common.Extensions

Reply:
removed